pkg/providers/version/version.go

```diff
@@ -51,14 +67,77 @@ func (p *Provider) Get(ctx context.Context) (string, error) {
 	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
 		return version.(string), nil
 	}
-	serverVersion, err := p.kubernetesInterface.Discovery().ServerVersion()
-	if err != nil {
-		return "", err
+	var version string
+	var err error
+	// If we're running locally, these environment variables will be empty.
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
```

Comment:
Nice! Auto-discovery! These will always be set in-cluster? Docs to it?

Reply:
Yeah, specifically this is to figure out if we're running within a pod in the same cluster that Karpenter's managing: https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#directly-accessing-the-rest-api , similar to the InClusterConfig().